pyproject.toml

```diff
@@ -12,3 +12,8 @@ profile = "black"
 multi_line_output = 3
 include_trailing_comma = true
 reverse_relative = true
+
+[tool.codespell]
+skip = '.git,*.pdf,*.svg'
+#
+# ignore-words-list = ''
```

Comment:
```suggestion
# ignore-words-list = ''
ignore-regex = "\bassertIn\b"
```

For case sensitive matching IIRC (adding whole on mobile)

Reply:
I'm just wondering if this will always be something I have to chase. How does it deal with function names, which might be out of my control if they don't match?

Reply:
there is such a common fear ;) in the projects using CamelCase or some other mixedCase, I just add regex to catch those, smth like (pick your poison)

```shell
❯ grep ignore-regex.*A-Z */.codespellrc
alphafold/.codespellrc:ignore-regex = (^\s*"image/\S+": ".*|\b[A-Z]+\b)
citation-file-format/.codespellrc:ignore-regex = ["`][A-Z][A-Z]["`]
gdcm-gdcm/.codespellrc:ignore-regex = \b([A-Z]+|[A-Z][a-z]*)\b
hpc-toolkit/.codespellrc:ignore-regex = HETATM -e CONECT|\([A-Z]\)[a-z]+
hugo/.codespellrc:ignore-regex = .*TE\?MP.*|.*codespell-ignore.*|[A-Z]?[a-z]*[A-Z][a-zA-Z]*|/extras/crossreferences/|Januar 2023|\\ndoes|.*ä.*
lazydocker/.codespellrc:ignore-regex = (\b[A-Za-z][a-z]*[A-Z]\S+\b|\.edn\b|\S+…|\\nd\b)
lazygit/.codespellrc:ignore-regex = (\b[A-Za-z][a-z]*[A-Z]\S+\b|\.edn\b|\S+…|\\nd\b)
NeuroML-Documentation/.codespellrc:ignore-regex = ^\s*"image/\S+": ".*|\b[a-z]*[A-Z][a-zA-Z]*\b
qlever/.codespellrc:ignore-regex = \b([A-Z]*[a-z]+[A-Z][a-zA-Z]*)\b|.*(Lorem ipsum|eleifend|feugait|codespell-ignore).*
quarto/.codespellrc:ignore-regex = \b[a-z]+[A-Z][A-Za-z]*\b
remark/.codespellrc:ignore-regex = \b(Hart|[A-Z]+)\b|.{300,}.*
saunafs/.codespellrc:ignore-regex = \b[A-Z]*[a-z]+[A-Z][A-Za-z]*\b|/mnt/[a-z0-9]+|.*\bcodespell-ignore\b.*
scidata/.codespellrc:ignore-regex = .*[úáüéè].*|\b[A-Z]+\b|\bvalue=".."
webknossos/.codespellrc:ignore-regex = \b(Manuel|[a-z]+[A-Z][a-zA-Z]*|H Mattern|Nat Commun|couldn&apos;t)\b
```

but as you can see here it was just a single hit, so I personally would not sweat for now. ;-)  generally it is better to avoid names (variables, arguments, or functions alike) coming close to typos.